proto/zitadel/user/v3alpha/authenticator.proto

```diff
@@ -0,0 +1,468 @@
+  // If the user has set a password, the time it was last changed will be returned.
+  Password password = 2;
+  // Meta information about the user's WebAuthN authenticators.
+  repeated WebAuthN web_auth_n = 3;
+  // A list of the user's time-based one-time-password (TOTP) authenticators,
+  // incl. the name for identification.
+  repeated TOTP totps = 4;
+  // A list of the user's one-time-password (OTP) SMS authenticators.
+  repeated OTPSMS otp_sms = 5;
+  // A list of the user's one-time-password (OTP) Email authenticators.
+  repeated OTPEmail otp_email = 6;
+  // A list of the user's authentication keys. They can be used to authenticate e.g. by JWT Profile.
+  repeated AuthenticationKey authentication_keys = 7;
+  // A list of the user's linked identity providers (IDPs).
+  repeated IdentityProvider identity_providers = 8;
+}
+
+message Username {
+  // unique identifier of the username.
+  string username_id = 1;
+  // The user's unique username. It is used for identification during authentication.
+  string username = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default usernames must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 3;
+}
+
+message SetUsername {
+  // Set the user's username. This will be used for identification during authentication.
+  string username = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1,
+      max_length: 200,
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default username must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 2;
+}
+
+message Password {
+  // States the time the password was last changed.
+  //(--TODO:do we want to include that in the list or is there a separate call?--)
+  google.protobuf.Timestamp last_changed = 1 [
```

Comment:
do we want to include that in the list or is there a separate call?

Reply:
To search for before and after a specific time?

Reply:
there have been some requests to be able to know it for a password age policy (since we don't implement it our own)

Reply:
Makes sense as we don't have a password age policy anymore then, I would include it in the list.